sources/packages/backend/apps/api/src/route-controllers/dynamic-form/dynamic-form.controller.ts

```diff
@@ -23,12 +28,46 @@ export class DynamicFormController extends BaseController {
     super();
   }
 
+  /**
+   * List all form definitions that contains the tag 'common' ordered by title.
+   * @returns list of form definitions.
+   */
+  @Roles(Role.AESTFormEditor)
   @Get()
-  async list(): Promise<any> {
-    return this.formService.list();
+  async list(): Promise<FormsAPIOutDTO> {
+    const allForms = await this.formService.list();
+    return {
+      forms: allForms.map((form) => ({
+        name: form.name,
+        title: form.title,
+      })),
+    };
   }
+
+  /**
+   * Get a form definition from Form.io.
+   * @param formName name of the form to be retrieved.
+   * @returns form definition.
+   */
   @Get(":formName")
   async getForm(@Param() formNameParam: FormNameParamAPIInDTO): Promise<any> {
     return this.formService.fetch(formNameParam.formName);
   }
+
+  /**
+   * Update a form definition in Form.io.
+   * @param formName Name of the form to be updated.
+   * @param payload Form definition to be updated.
+   */
+  @Roles(Role.AESTFormEditor)
+  @Put(":formName")
```

Comment:
Correct me if I am wrong. As per the form IO API specification it is the formPath and not the formName. As we have both as same it is working for us.

![image](https://github.com/user-attachments/assets/634b6d1d-f95d-4874-a747-487eca4eaf49)

Reply:
It is the same everywhere else in the application. I will change the ones for the new implementation and see the impact of changing the existing ones, maybe in a separate PR.